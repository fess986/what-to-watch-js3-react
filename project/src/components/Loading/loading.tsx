type LoadingProps = {
  caller? : string;
}

const Loading = ({caller = ''} : LoadingProps) : JSX.Element => (
  <h1>{`Loading${caller}...`}</h1>
);

export default Loading;
