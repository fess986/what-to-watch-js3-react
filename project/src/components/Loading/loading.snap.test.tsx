import {render} from '@testing-library/react';
import Loading from './loading';

describe('testing Loading screen render', () => {
  it('Should render correctly', () => {

    const {container} = render(<Loading />); // рендерим компонент и помещаем в контейнер при помощи render от @testing-library/react.
    // <Loading count={count}/> - так же сюда можно какие-либо пропсы передавать, если того требует компонент

    expect(container).toMatchSnapshot();
  });
});
