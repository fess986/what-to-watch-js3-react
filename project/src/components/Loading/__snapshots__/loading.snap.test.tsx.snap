// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Loading screen render Should render correctly 1`] = `
<div>
  <h1>
    Loading...
  </h1>
</div>
`;
